packages/builder/src/stores/builder/rowActions.ts

```diff
@@ -34,26 +45,30 @@ export class RowActionStore extends BudiStore {
 
     // Fetch row actions for this table
     const res = await API.rowActions.fetch(tableId)
-    const actions = Object.values(res || {})
+    const actions = Object.values(res || {}) as RowAction[]
     this.update(state => ({
       ...state,
       [tableId]: actions,
     }))
   }
 
-  createRowAction = async (tableId, viewId, name) => {
+  createRowAction = async (tableId: string, viewId?: string, name?: string) => {
     if (!tableId) {
       return
     }
 
     // Get a unique name for this action
     if (!name) {
-      const existingRowActions = get(this.store)[tableId] || []
+      const existingRowActions = get(this)[tableId] || []
       name = getSequentialName(existingRowActions, "New row action ", {
         getName: x => x.name,
       })
     }
 
+    if (!name) {
```

Comment:
Is this potentially an error scenario?

Reply:
You're right, let's be more explicit and actually return an error here. I can't see a case in which it gets here, something would need to break in the `getSequentialName` function which is itself a fallback. But will throw here just to be safe